src/Xrefcheck/Verify.hs

```diff
@@ -768,7 +768,7 @@ checkExternalResource followed config@Config{..} link
           other -> throwError $ ExternalResourceSomeError $ show other
       where
         retryAfterInfo :: Response a -> Maybe RetryAfter
-        retryAfterInfo = readMaybe . decodeUtf8 <=< L.lookup hRetryAfter . responseHeaders
+        retryAfterInfo = readMaybe . decodeUtf8 @Text <=< L.lookup hRetryAfter . responseHeaders
```

Comment:
In this case, `readMaybe` comes from `Text.Read` and takes a `String` parameter, so it does not compile for me unless this type parameter application is removed.

Reply:
I was surprised that I had to add it, because it is orthogonal to the concern of this PR, namely `optparse-applicative`. 
Let me investigate some more...

Reply:
So indeed `stack build` fails now (which uses LTS 19 which is GHC 9.0), but `cabal build -w ghc-9.2.8` succeeds and needs this change.  
I am gonna remove this patch, it is something to be sorted out when migrating from GHC 9.0 to 9.2 and up.
(Note that GHC 9.2.8 is the currently recommended version of GHC.)

Reply:
Yes, it should be updated at some point. I will create an issue referencing this comments thread. Thank you!

Reply:
NB: `cabal build -w ghc-9.0.2` fails because of:
- https://github.com/vincenthz/hs-connection/issues/54